### PR TITLE
Statistics messages for ROS 2 topic statistics and system metrics

### PR DESCRIPTION
**Pull request by prajakta-gokhale**
Related to https://github.com/ros-tooling/aws-roadmap/issues/249

* Move `metrics_statistics_msgs` into this repo, out of `system_metrics_collector`
* Add lint and test actions for new repo

Signed-off-by: Prajakta Gokhale <prajaktg@amazon.com>

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      fail-fast: false
+      matrix:
+          os: [macOS-latest, ubuntu-18.04]
+    steps:
+    - uses: ros-tooling/setup-ros@0.0.19
+    - uses: ros-tooling/action-ros-ci@0.0.15
+      with:
+        package-name: metrics_statistics_msgs
+        colcon-mixin-name: coverage-gcc
+        colcon-mixin-repository: https://raw.githubusercontent.com/colcon/colcon-mixin-repository/master/index.yaml
```
Comment (thomas-moulard) [inline .github/workflows/test.yml:25]: do we need coverage for messages?

File: README.md [hunk 2]
```diff
@@ -1,3 +1,16 @@
 # metrics_statistics_msgs
 
 Package containing ROS 2 message definitions for reporting statistics for topics and system resources.
+
+![License](https://img.shields.io/github/license/ros-tooling/system_metrics_collector)
+[![GitHub Action Status](https://github.com/ros-tooling/metrics_statistics_msgs/workflows/Test%20metrics_statistics_msgs/badge.svg)](https://github.com/ros-tooling/metrics_statistics_msgs/actions?query=workflow%3A%22Test+metrics_statistics_msgs%22)
+
+### Messages defined in this package
+1. StatisticDataType
```
Comment (thomas-moulard) [inline README.md:9]: missing newline, please run `mdlint` as a quick check

File: metrics_statistics_msgs/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+# Enable strict compiler flags if possible.
+include(CheckCXXCompilerFlag)
+# Removed -Wmissing-declarations until gmock is ignored
+set(FLAGS -pedantic -Wno-long-long -Wall -Wextra -Wcast-qual -Wformat -Wwrite-strings -Wcast-align -Wno-error=cast-align)
+foreach(FLAG ${FLAGS})
+  check_cxx_compiler_flag(${FLAG} R${FLAG})
+  if(${R${FLAG}})
+    set(WARNING_CXX_FLAGS "${WARNING_CXX_FLAGS} ${FLAG}")
+  endif()
+endforeach()
```
Comment (thomas-moulard) [inline metrics_statistics_msgs/CMakeLists.txt:23]: not meaningful here

File: metrics_statistics_msgs/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+# Default to C99
+if(NOT CMAKE_C_STANDARD)
+  set(CMAKE_C_STANDARD 99)
+endif()
+
+# Default to C++14
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
+endif()
```
Comment (thomas-moulard) [inline metrics_statistics_msgs/CMakeLists.txt:12]: useful?

File: metrics_statistics_msgs/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+  if(${R${FLAG}})
+    set(WARNING_CXX_FLAGS "${WARNING_CXX_FLAGS} ${FLAG}")
+  endif()
+endforeach()
+
+# find dependencies
+find_package(ament_cmake REQUIRED)
+find_package(std_msgs REQUIRED)
+find_package(builtin_interfaces REQUIRED)
+find_package(rosidl_default_generators REQUIRED)
```
Comment (thomas-moulard) [inline metrics_statistics_msgs/CMakeLists.txt:29]: alphasort

File: metrics_statistics_msgs/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+set(msg_files
```
Comment (thomas-moulard) [inline metrics_statistics_msgs/CMakeLists.txt:31]: no need to define a variable for something used only once

File: metrics_statistics_msgs/CMakeLists.txt (hunk 4 above)
Comment (piraka9011) [reply metrics_statistics_msgs/CMakeLists.txt:12]: This is the default when creating a new package.
But given this is a msgs only package, probably not.

File: .github/workflows/test.yml (hunk 1 above)
Comment (dabonnie) [reply .github/workflows/test.yml:25]: I haven't seen coverage used for other message repos, but if there's a good example we could follow that.

File: README.md [hunk 7]
```diff
@@ -1,3 +1,16 @@
...
 Package containing ROS 2 message definitions for reporting statistics for topics and system resources.
+
+![License](https://img.shields.io/github/license/ros-tooling/system_metrics_collector)
+[![GitHub Action Status](https://github.com/ros-tooling/metrics_statistics_msgs/workflows/Test%20metrics_statistics_msgs/badge.svg)](https://github.com/ros-tooling/metrics_statistics_msgs/actions?query=workflow%3A%22Test+metrics_statistics_msgs%22)
+
+### Messages defined in this package
+1. StatisticDataType  
+A message that represent the types of statistics reported. This includes `average`, `minimum`, `maximum`, `stddev` and `sample_count`.
+
+2. StatisticDataPoint
```
Comment (dabonnie) [inline README.md:12]: Maybe an example of how to use this? Currently not explicit that data_type maps the reported statistics type.

File: README.md [hunk 8]
```diff
@@ -1,3 +1,16 @@
...
+
+### Messages defined in this package
+1. StatisticDataType  
+A message that represent the types of statistics reported. This includes `average`, `minimum`, `maximum`, `stddev` and `sample_count`.
+
+2. StatisticDataPoint  
+A message that represents a single statistic value. This includes `data_type` and `data` value.
+
+3. MetricsMessage  
+A message that represents statistic data measured from a source within a time window. This includes `measurement_source_name`, `metrics_source`, `unit`, `window_start`, `window_stop` and `statistics`.
```
Comment (dabonnie) [inline README.md:16]: Explicitly note what `statistics` represents

File: metrics_statistics_msgs/package.xml [hunk 9]
```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0"?>
+<?xml-model href="http://download.ros.org/schema/package_format3.xsd" schematypens="http://www.w3.org/2001/XMLSchema"?>
+<package format="3">
+  <name>metrics_statistics_msgs</name>
+  <version>0.0.0</version>
+  <description>Message definition to publish system and message metrics.</description>
```
Comment (dabonnie) [inline metrics_statistics_msgs/package.xml:6]: Should we reference topics statistics here as well?

File: metrics_statistics_msgs/CMakeLists.txt (hunk 6 above)
Comment (emersonknapp) [reply metrics_statistics_msgs/CMakeLists.txt:31]: There's something to be said for encapsulating the list separate from the rest of the arguments to the function. I've seen this pattern a lot and think it's fine

File: .github/workflows/test.yml (hunk 1 above)
Comment (prajakta-gokhale) [reply .github/workflows/test.yml:25]: I don't think we'll need it so will remove.

File: metrics_statistics_msgs/CMakeLists.txt (hunk 6 above)
Comment (prajakta-gokhale) [reply metrics_statistics_msgs/CMakeLists.txt:31]: This follows the pattern elsewhere in ros2. [std_msgs](https://github.com/ros2/common_interfaces/blob/master/std_msgs/CMakeLists.txt) for example.

File: metrics_statistics_msgs/package.xml (hunk 9 above)
Comment (prajakta-gokhale) [reply metrics_statistics_msgs/package.xml:6]: Isn't message metrics same as topic statistics? I can reword it to topic statistics.

File: metrics_statistics_msgs/CMakeLists.txt (hunk 4 above)
Comment (prajakta-gokhale) [reply metrics_statistics_msgs/CMakeLists.txt:12]: Right, this was the default when the package was created. Removing since it isn't needed.

File: README.md (hunk 8 above)
Comment (prajakta-gokhale) [reply README.md:16]: Added explanation for all message fields.

File: README.md (hunk 7 above)
Comment (prajakta-gokhale) [reply README.md:12]: Added explanation for each field in the message, which should make this clear.

File: metrics_statistics_msgs/package.xml [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+  <version>0.0.0</version>
+  <description>Message definition to publish system metrics topic statistics.</description>
+  <maintainer email="ros-tooling@googlegroups.com">ROS Tooling Working Group</maintainer>
+  <license>Apache License 2.0</license>
+
+  <buildtool_depend>ament_cmake</buildtool_depend>
+  <buildtool_depend>rosidl_default_generators</buildtool_depend>
+
+  <build_depend>builtin_interfaces</build_depend>
+  <build_depend>std_msgs</build_depend>
```
Comment (thomas-moulard) [inline metrics_statistics_msgs/package.xml:14]: not needed?
